test-clients/src/main/java/com/hedera/services/bdd/suites/token/TokenUpdateSpecs.java

```diff
@@ -236,6 +238,43 @@ public HapiApiSpec symbolChanges() {
 				);
 	}
 
+	public HapiApiSpec nameChanges() {
+		var hopefullyUnique = "ORIGINAL" + TxnUtils.randomUppercase(5);
+
+		return defaultHapiSpec("NameChanges")
+				.given(
+						newKeyNamed("adminKey"),
+						cryptoCreate(TOKEN_TREASURY)
+				).when(
+						tokenCreate("tbu")
+								.adminKey("adminKey")
+								.treasury(TOKEN_TREASURY),
+						tokenUpdate("tbu")
+								.name(hopefullyUnique)
+				).then(
+						getTokenInfo("tbu").hasName(hopefullyUnique)
+				);
+	}
+
+	public HapiApiSpec nameChangeFails() {
```

Comment:
- IMO a better name for the test could be: `tooLongNameCheckHolds`.
- `emptyName` is not used as a variable - delete it
- IMO you should create a constant similar to `TOKEN_TREASURY` for the `MAX_NAME_LENGTH`. If for some reason we change that value, it would be better to have only one or two places that we need to change the value so that the tests are working again. You can introduce a new const variable in the TokenCreateSpecs as-well.